### PR TITLE
Adding support for `squeeze` and `unsqueeze`

### PR DESCRIPTION
**Pull request by mtopalovicTT**
Adding `squeeze` and `unsqueeze` tests
Added support for squeeze unsqueeze in forge
Correctly calculate shape of tensor for unsqueeze when negative dim is passed

### CODE REVIEW

File: forge/test/mlir/test_ops.py [hunk 1]
```diff
@@ -89,6 +89,66 @@ def forward(self, a):
     co_out = [co.to("cpu") for co in co_out]
     assert compare_with_golden_pcc(golden=fw_out, calculated=co_out[0], pcc=0.99)
 
+@pytest.mark.parametrize("input_shape_and_dim",
```
[the remark this answers is not in the document]
Comment (mtopalovicTT) [reply forge/test/mlir/test_ops.py:92]: Shouldn't golden_pcc check that?

File: forge/test/mlir/test_ops.py [hunk 2]
```diff
@@ -89,6 +89,66 @@ def forward(self, a):
...
+    inputs = [torch.rand(*input_shape)]
+
+    framework_model = Unsqueeze()
+    fw_out = framework_model(*inputs)
+
+    compiled_model = forge.compile(framework_model, sample_inputs=inputs)
+    co_out = compiled_model(*inputs)
+
+    co_out = [co.to("cpu") for co in co_out]
+    assert compare_with_golden_pcc(golden=fw_out, calculated=co_out[0], pcc=0.99)
```
[the remark this answers is not in the document]
Comment (mtopalovicTT) [reply forge/test/mlir/test_ops.py:150]: Shouldn't golden_pcc check that?

[the remark this answers is not in the document]
Comment (mtopalovicTT) [reply forge/test/mlir/test_ops.py:152]: Added

**Comment by mtopalovicTT** [comment]
@chandrasekaranpradeep @nvukobratTT Removed decompose of unsqueeze and squeeze. Please review.

File: forge/forge/tvm_to_python.py [hunk 3]
```diff
@@ -810,6 +810,44 @@ def populate_vstack_args(graph, nid, compiler_cfg):
     args = [("slices", f"{slice_size}")]
     return args
 
+def populate_unsqueeze_args(graph, nid, compiler_cfg):
+    node = graph["nodes"][nid]
+    input_shape = graph["nodes"][node["inputs"][0][0]]["attrs"]["shape"][0][0]
+    output_shape = node["attrs"]["shape"][0][0]
+    
+    assert len(input_shape) + 1 == len(output_shape), "Unsqueeze only supports adding 1 dimension"
```
Comment (nvukobratTT) [inline forge/forge/tvm_to_python.py:818]: Do we maybe have dim as part of the node attributes? 

If yes, we can simplify this check and below logic for determining which dim is requested. Same applies for squeeze args as well.

File: forge/forge/tvm_to_python.py [hunk 4]
```diff
@@ -810,6 +810,44 @@ def populate_vstack_args(graph, nid, compiler_cfg):
...
+
+    dim = -1
+    for i, (inp_dim, out_dim) in enumerate(zip(input_shape, output_shape)):
+        if inp_dim != out_dim:
+            dim = i
+            break
+
+    assert dim != -1
```
Comment (nvukobratTT) [inline forge/forge/tvm_to_python.py:826]: Do we support negative (un)squeeze dims then? Also, let's add comments to all asserts just to give user a bit more details

File: forge/test/mlir/test_ops.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nvukobratTT) [reply forge/test/mlir/test_ops.py:92]: Besides, let's explore testing with those negative dims as well. I think we should be able to support this part as well

**Review by nvukobratTT — APPROVED** [review]
LGTM! 🙌

**Pull request merged**